src/main.ts

```diff
@@ -256,7 +258,14 @@ export default class ExecuteCodePlugin extends Plugin {
 				console.log(`runCodeInShell ${this.settings.csPath} ${this.settings.csArgs} ${"cs"}`)
 				this.runCodeInShell(transformedCode, out, button, this.settings.csPath, this.settings.csArgs, "csx");
 			});
-
+      
+		} else if (language.contains("language-haskell") || language.contains("language-hs")) {
+			button.addEventListener("click", async () => {
+				button.className = runButtonDisabledClass;
+				const transformedCode = await this.injectCode(srcCode, "haskell");
```

Comment:
Since you opened your PR, the `injectCode` method was refactored into a different class. If you could please update this line, that would be great! :)
Try:
```ts
const transformedCode = await new CodeInjector(this.app, this.settings, "haskell").injectCode(srcCode);
```